File: packages/tasks-gen/snippets-fixtures/conversational-llm-stream/js/huggingface.js/0.together.js
import { InferenceClient } from "@huggingface/inference";

const client = new InferenceClient("api_token");

let out = "";

const stream = client.chatCompletionStream({
    provider: "together",
    model: "meta-llama/Llama-3.1-8B-Instruct",
    messages: [
        {
            role: "user",
            content: "What is the capital of France?",
        },
    ],
});

for await (const chunk of stream) {
	if (chunk.choices && chunk.choices.length > 0) {
		const newContent = chunk.choices[0].delta.content;
		out += newContent;
		console.log(newContent);
	}
}

File: packages/tasks-gen/snippets-fixtures/conversational-llm-stream/python/huggingface_hub/0.together.py
from huggingface_hub import InferenceClient, get_token

client = InferenceClient(
    api_key=get_token(),
)

stream = client.chat.completions.create(
    model="meta-llama/Llama-3.1-8B-Instruct:together",
    messages=[
        {
            "role": "user",
            "content": "What is the capital of France?"
        }
    ],
    stream=True,
)

for chunk in stream:
    print(chunk.choices[0].delta.content, end="")